swift/Workflow/Sources/Lifetime.swift

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+public final class Lifetime {
+    public func onEnded(_ action: @escaping () -> Void) {
```

Comment:
Does this relate to any existing behavior in RxSwift, ReactiveSwift, or Combine? I'm trying to understand if this is a "Completion" event.

Reply:
Yup, this is pretty much a completion event. 

This behavior exists in each of those libraries. (`Disposable`, `Cancellable`, `Lifetime`, ...etc)